src/app/file-manager/file-manager.module.ts

```diff
@@ -35,7 +44,37 @@ import { GcsService } from './services/gcs.service';
   ],
   providers: [
     FilesService,
-    GcsService
+    GcsApiService,
+    GcsApiMockService,
+    FirecloudApiService,
+    FirecloudApiMockService,
+    {
+      provide: HTTP_INTERCEPTORS,
+      useClass: RequestInterceptor,
+      multi: true
+    },
+    {
+      provide: FirecloudService,
+      deps: [HttpClient],
+      useFactory(http: HttpClient) {
+        if (environment.testing) {
+          return new FirecloudApiMockService(http);
```

Comment:
Can you construct this without passing in `http`? This and `GcsApiMockService` should be constructable without `HttpClient`.